Backpack/src/main/java/net/skyscanner/backpack/skeleton/BpkShimmerOverlay.kt

```diff
@@ -0,0 +1,69 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.skeleton
+
+import android.animation.ObjectAnimator
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.FrameLayout
+import androidx.annotation.AttrRes
+import net.skyscanner.backpack.R
+
+class BpkShimmerOverlay @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  @AttrRes defStyleAttr: Int = 0,
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+  private var inflatingInProgress = false
+  init {
+    inflatingInProgress = true
+    inflate(this.context, R.layout.bpk_shimmer_skeleton, this)
+    inflatingInProgress = false
+    startShimmer()
+  }
+
+  override fun addView(child: View?) {
+    if (inflatingInProgress) {
```

Comment:
you could take a similar approach to `onViewAdded` where you check if the view being added has the `R.id.bpk_skeleton_container` id instead of keeping a boolean value here for inflation

Reply:
this is nice, thanks.